keras/engine/topology.py

```diff
@@ -1331,9 +1331,10 @@ def compute_mask(self, inputs, mask=None):
             masks = [K.expand_dims(m, 0) for m in mask if m is not None]
             return K.all(K.concatenate(masks, axis=0), axis=0, keepdims=False)
         elif self.mode == 'concat':
-            masks = [K.ones_like(inputs[i][:-1]) if m is None else m for i, m in zip(inputs, mask)]
-            expanded_dims = [K.expand_dims(m) for m in masks]
-            concatenated = K.concatenate(expanded_dims, axis=self.concat_axis)
+            # Make sure mask has the same number of dimensions as the input
+            expand_if_needed = lambda input, mask: K.expand_dims(mask) if K.ndim(mask) < K.ndim(input) else mask
```

Comment:
In which cases would this be needed?

It may be better to defined a proper private function than a very long named lambda.

e.g.

```python
def _expand_if_needed(input, mask):
   ...
```